src/main/java/org/opensearch/ad/AnomalyDetectorProfileRunner.java

```diff
@@ -290,26 +300,69 @@ private void prepareProfile(
 
     private void profileEntityStats(MultiResponsesDelegateActionListener<DetectorProfile> listener, AnomalyDetector detector) {
         List<String> categoryField = detector.getCategoryField();
-        if (categoryField == null || categoryField.size() != 1) {
+        if (categoryField == null || categoryField.size() == 0 || categoryField.size() > NumericSetting.maxCategoricalFields()) {
```

Comment:
Can we use `detector.isMultientityDetector` here?

Reply:
changed